beacon_node/beacon_chain/src/validator_custody.rs

```diff
@@ -170,7 +184,30 @@ impl CustodyContext {
                     .into_iter()
                     .collect(),
             }),
+            all_custody_groups_ordered: OnceLock::new(),
+            _phantom_data: PhantomData,
+        }
+    }
+
+    pub fn init_all_custody_groups_ordered(
+        &self,
+        all_custody_groups_ordered: Vec<CustodyIndex>,
+        spec: &ChainSpec,
+    ) -> Result<(), String> {
+        let mut ordered_custody_groups = vec![];
```

Comment:
I don't understand this.
We basically want to store an ordering of data columns for a given `NodeId`.
Why are we storing columns for every custody index here? Can't we just compute the ordering for the node id and take a slice [..cgc] for the required cgc value?

Reply:
yeah, this is due to custody groups decoupling - because number of columns could be different to custody groups. The alternative is storing all columns instead of groups, but it may not be as ergonomic because all the configs deal with groups (`sample_per_slot`,  `custody_requirement` etc)

`cgc` doesn't directly translate to columns either, we'd need to get the `sampling_size` (which also uses custody groups as unit), and then convert it to number of columns.

I think it's also doable though - so we'd
- store an ordered list of all columns
- when CGC changes, compute the `sampling_size` and then convert it to `sampling_size_columns`
- and take a slice of all columns using `sampling_size_columns`

Reply:
aghhh i hate the unnecessary abstraction with custody groups. I'm willing to bet that we will never change the columns per custody group and this will just be a painful tech debt forever 😭 

I also prefer having the columns to sample in one place to avoid syncing up everything if possible. 
I really like the approach of storing an ordered list of columns that you described. The computation is constant time so shouldn't be too expensive

Reply:
yeah the custody groups abstraction complicates a lot of things, including the tests.

I've updated it to store an ordered list of columns, i think it seems to work out quite well, and also managed to get rid of `network_globals.sampling_columns`, which allowed us to remove `globals` from a bunch of places 🎉